tangelo/toolboxes/operators/trim_trivial_qubits.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from tangelo.toolboxes.operators import QubitOperator
+from tangelo.linq import Circuit
+from tangelo.linq.helpers.circuits import pauli_string_to_of, pauli_of_to_string
+
+
+def trim_trivial_operator(qu_op, n_qubits, trim_index, trim_states, reindex=True):
+    """
+    Calculates expectation values of a QubitOperator acting on qubits in a
+    trivial |0> or |1> state. Returns a trimmed QubitOperator with updated coefficients
+
+    Args:
+        qu_op (QubitOperator): Operator to trim
+        n_qubits (int): number of qubits in full system
+        trim_index (list):  index of qubits to trim
+        trim_states (list): state of the qubits to trim, 0 or 1
+        reindex (bool): If True, remaining qubits will be reindexed
+    Returns:
+        qu_op  (QubitOperator): trimmed QubitOperator with updated coefficients
+    """
+    trim_states = [x for (y, x) in sorted(zip(trim_index, trim_states), key=lambda pair: pair[0])]
+    trim_index = sorted(trim_index)
```

Comment:
Yeah that probably would make things easier to work with, I'll reformat!
I wasn't sure whether it would be easier for the user to input these as two lists or as a dictionary.